City-of-Helsinki/events-helsinki-monorepo
# LIIKUNTA-584 | feat: upgrade to HDS v3.1.0 and HCRC that uses HDS v3.1.0 & update tests

## Description

**NOTE**:
 - The **visual** outlook has been **approved** by the **designer**

### feat: upgrade to HDS v3.1.0 and HCRC that uses HDS v3.1.0 & update tests

- upgrade to hds-react 3.1.0
- upgrade to react-helsinki-headless-cms 1.0.0-alpha229
- UI components:
  - `<Footer>` was rebuilt in HDS v3 so needed to make changes to it
  - rename flipHorizontal to flipVertical per HDS v3 release notes
- tests:
  - update snapshots
	- NOTE: **snapshot updating** updated **suspiciously** e.g.
	  "tag_hds-tag__label__I6mc8" to **"undefined"**, maybe there's
	  something broken?
  - fix browser tests because language selector changed in HCRC's
	Navigation component which is now actually using HDS v3's Header
	component underneath
- storybook:
  - fix "yarn build-storybook" and "yarn storybook" failing under
	packages/components
	- NOTE: the storybook looks broken but at least it builds and runs
  - add "crypto-browserify" because it was needed by HDS v3
  - fix graphql code generation creating unusable enumeration values
	by making sure the enum keys don't start with a digit

See react-helsinki-headless-cms 1.0.0-alpha229 PR:
https://github.com/City-of-Helsinki/react-helsinki-headless-cms/pull/131

refs LIIKUNTA-584

## Issues

### Closes

[LIIKUNTA-584](https://helsinkisolutionoffice.atlassian.net/browse/LIIKUNTA-584)

### Related

## Testing

### Automated tests

### Manual testing

## Screenshots

![image](https://github.com/City-of-Helsinki/events-helsinki-monorepo/assets/77663720/bd5e591d-257f-4a03-85c5-6b91cd5b51c0)

## Additional notes


[LIIKUNTA-584]: https://helsinkisolutionoffice.atlassian.net/browse/LIIKUNTA-584?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
